lms/static/js/groups/views/cohort_editor.js

```diff
@@ -39,6 +40,10 @@ var edx = edx || {};
             return this;
         },
 
+        updateHeaderTitle: function() {
```

Comment:
Pull the cohort-management-group-header header section into its own template so that you can simply re-render that template on Save. This will also fix the bug that the description does not update.